cli/exchange/service.go

```diff
@@ -673,3 +673,50 @@ func GetOpYamlArchiveFromClusterDepl(deploymentConfig string) []byte {
 
 	return nil
 }
+
+type ServiceNode struct {
+	OrgId          string `json:"orgid"`
+	ServiceUrl     string `json:"serviceURL"`
+	ServiceVersion string `json:"serviceVersion"`
+	ServiceArch    string `json:"serviceArch"`
+}
+
+// List the nodes that a service is running on.
+func ListServiceNodes(org, userPw, svcId, nodeOrg string) {
+	//check for ExchangeUrl early on
+	var exchUrl = cliutils.GetExchangeUrl()
+
+	// if nodeOrg is not specified, default to service org
+	if nodeOrg == "" {
+		nodeOrg = org
+	}
+
+	// extract service id
+	var svcOrg string
+	svcOrg, svcId = cliutils.TrimOrg(org, svcId)
+
+	// get service from the Exchange
+	var services exchange.GetServicesResponse
+	httpCode := cliutils.ExchangeGet("Exchange", cliutils.GetExchangeUrl(), "orgs/"+svcOrg+"/services"+cliutils.AddSlash(svcId), cliutils.OrgAndCreds(org, userPw), []int{200, 404}, &services)
+	if httpCode == 404 {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, fmt.Sprintf("service id does not exist in the Exchange: %v/%v", svcOrg, svcId))
+	} else {
+		// extract org id, service url, version, and arch from Exchange
+		var svcNode ServiceNode
+		for id, _ := range services.Services {
+			svcNode = ServiceNode{OrgId: svcOrg, ServiceUrl: services.Services[id].URL, ServiceVersion: services.Services[id].Version, ServiceArch: services.Services[id].Arch}
+			break
+		}
+
+		// structure we are writing to
+		var listNodes map[string]interface{}
+		cliutils.ExchangePutPost("Exchange", http.MethodPost, exchUrl, "orgs/"+nodeOrg+"/search/nodes/service", cliutils.OrgAndCreds(org, userPw), []int{201}, svcNode, &listNodes)
+
+		// print list
+		jsonBytes, err := json.MarshalIndent(listNodes, "", cliutils.JSON_INDENT)
+		if err != nil {
+			cliutils.Fatal(cliutils.JSON_PARSING_ERROR, fmt.Sprintf("failed to marshal 'hzn exchange service list' output: %v", err))
```

Comment:
minor typo: message should read "failed to marshal 'hzn exchange service listnode' output:

Reply:
Done.